### PR TITLE
add redux saga typesafe-actions

### PR DESCRIPTION
**Pull request by Beker94**

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -2,5 +2,19 @@ import React from 'react';
...
+import { Provider } from 'react-redux';
+import { createStore, applyMiddleware } from 'redux';
+import createSagaMiddleware from 'redux-saga';
 
-ReactDOM.render(<App />, document.getElementById('root'));
+import { reducer } from './redux/reducer';
+
+const saga = createSagaMiddleware();
+
+const store = createStore(reducer, applyMiddleware(saga));
```
Comment (enheit) [inline src/index.tsx]: I would recommend to move `store` and `saga` init configuration into separate folders.
Let me show what I mean.

Create `/src/store` folder and put `store` variable into `root.store.ts` file and `saga` into `root.saga.ts` file.

At the end you should have something like this

```
src/
  store/
    root.store.ts
    root.saga.ts
```

File: src/index.tsx [hunk 2]
```diff
@@ -2,5 +2,19 @@ import React from 'react';
 import ReactDOM from 'react-dom';
 import './index.css';
 import App from './App';
+import { Provider } from 'react-redux';
+import { createStore, applyMiddleware } from 'redux';
+import createSagaMiddleware from 'redux-saga';
 
-ReactDOM.render(<App />, document.getElementById('root'));
+import { reducer } from './redux/reducer';
```
Comment (enheit) [inline src/index.tsx]: Also move reducer to the `src/store/root.reducer.ts`

**Review by Pashkaua — APPROVED** [review]
Good job !

**Review by enheit — APPROVED** [review]

**Review by Danik1702 — APPROVED** [review]
Very well!

**Pull request merged**